cli branch (#13880)
pingcap/docs: tidb-cloud/ticloud-branch-create.md

---
title: ticloud branch create
summary: The reference of `ticloud branch create`.
---

# ticloud branch create

> Create a branch:

Reviewer: ```suggestion
Create a branch for a cluster:
```

```shell
ticloud branch create [flags]
```

> **Note:**
>

> > Currently, you can only create branches for TiDB Serverless cluster.

Reviewer 2: ```suggestion
> Currently, you can only create branches for [TiDB Serverless](/tidb-cloud/select-cluster-tier.md#tidb-serverless) cluster.
```

> > Currently, you can only create branch under [TiDB Serverless](/tidb-cloud/select-cluster-tier.md#tidb-serverless-beta).

Reviewer 3: better change a link. beta will be removed I think

> > Currently, you can only create branch under [TiDB Serverless](/tidb-cloud/select-cluster-tier.md#tidb-serverless-beta).

Reviewer: ```suggestion
> Currently, you can only create branches for [TiDB Serverless](/tidb-cloud/select-cluster-tier.md#tidb-serverless-beta) clusters.
```

## Examples

Create a branch in interactive mode:

```shell
ticloud branch create
```

Create a branch in non-interactive mode:

```shell
ticloud branch create --cluster-id <cluster-id> --branch-name <branch-name>
```

## Flags

In non-interactive mode, you need to manually enter the required flags. In interactive mode, you can just follow CLI prompts to fill them in.

| Flag                    | Description                                                | Required | Note                                                |
|-------------------------|------------------------------------------------------------|----------|-----------------------------------------------------|
| -c, --cluster-id string | The ID of the cluster, in which the branch will be created | Yes      | Only works in non-interactive mode.                 |

> | --branch-name string    | Name of the branch to be created                           | Yes      | Only works in non-interactive mode.                 |

Reviewer: ```suggestion
| --branch-name string    | The name of the branch to be created                           | Yes      | Only works in non-interactive mode.                 |
```
